File: 2239. Find Closest Number to Zero/2239. Find Closest Number to Zero.java
class Solution {

  public int findClosestNumber(int[] nums) {
    int closest = nums[0];

    for (int num : nums) {
      if (Math.abs(num) < Math.abs(closest)) {
        closest = num;
      }
    }

    if (closest < 0 && contains(nums, Math.abs(closest))) {
      closest *= -1;
    }

    return closest;
  }

  private boolean contains(int[] nums, int closest) {
    for (int i = 0; i < nums.length; i++) {
      if (closest == nums[i]) {
        return true;
      }
    }
    return false;
  }
}

File: 2239. Find Closest Number to Zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        closest = float("inf")
        for n in nums:
            if abs(n) < abs(closest):
                closest = n

        if closest < 0 and abs(closest) in nums:
            closest *= -1

        return closest
